crates/wasmtime/src/config.rs

```diff
@@ -2003,47 +2003,180 @@ impl PoolingAllocationConfig {
+    ///
+    /// If a component will instantiate more core instances than `count`, then
+    /// the component will fail to instantiate.
+    pub fn max_core_instances_per_component(&mut self, count: u32) -> &mut Self {
+        self.config.limits.max_core_instances_per_component = count;
+        self
+    }
+
+    /// The maximum number of Wasm linear memories that a single component may
+    /// transitively contain (default is `20`).
+    ///
+    /// This method (along with
+    /// [`PoolingAllocationConfig::max_core_instances_per_component`],
+    /// [`PoolingAllocationConfig::max_tables_per_component`], and
+    /// [`PoolingAllocationConfig::component_instance_size`]) allows you to cap
+    /// the amount of resources a single component allocation consumes.
+    ///
+    /// If a component transitively contains more linear memories than `count`,
+    /// then the component will fail to instantiate.
+    pub fn max_memories_per_component(&mut self, count: u32) -> &mut Self {
+        self.config.limits.max_memories_per_component = count;
+        self
+    }
+
+    /// The maximum number of tables that a single component may transitively
+    /// contain (default is `20`).
+    ///
+    /// This method (along with
+    /// [`PoolingAllocationConfig::max_core_instances_per_component`],
+    /// [`PoolingAllocationConfig::max_memories_per_component`],
+    /// [`PoolingAllocationConfig::component_instance_size`]) allows you to cap
+    /// the amount of resources a single component allocation consumes.
+    ///
+    /// If a component will transitively contains more tables than `count`, then
+    /// the component will fail to instantiate.
+    pub fn max_tables_per_component(&mut self, count: u32) -> &mut Self {
+        self.config.limits.max_tables_per_component = count;
+        self
+    }
+
+    /// The maximum number of concurrent Wasm linear memories supported (default
+    /// is `1000`).
     ///
     /// This value has a direct impact on the amount of memory allocated by the pooling
     /// instance allocator.
     ///
-    /// The pooling instance allocator allocates three memory pools with sizes depending on this value:
+    /// The pooling instance allocator allocates a memory pool, where each entry
+    /// in the pool contains the reserved address space for each linear memory
+    /// supported by an instance.
+    ///
+    /// The memory pool will reserve a large quantity of host process address
+    /// space to elide the bounds checks required for correct WebAssembly memory
+    /// semantics. Even with 64-bit address spaces, the address space is limited
+    /// when dealing with a large number of linear memories.
+    ///
+    /// For example, on Linux x86_64, the userland address space limit is 128
+    /// TiB. That might seem like a lot, but each linear memory will *reserve* 6
+    /// GiB of space by default.
+    pub fn total_memories(&mut self, count: u32) -> &mut Self {
```

Comment:
as above, `max_...` ?

Reply:
This one isn't really a maximum, it is the exact number of memories we will have capacity for in the pool. So I'll leave it as-is.

Reply:
Hmm, yeah, I guess it's kind of a matter of semantics -- it *is* the maximum number that we can allocate across the whole pool. ("Capacity" is another way of saying "maximum", I guess.) But I'll defer to your judgment here if you want to keep as-is.